execution/gethexec/executionengine.go

```diff
@@ -486,15 +502,15 @@ func (s *ExecutionEngine) ResultAtPos(pos arbutil.MessageIndex) (*execution.Mess
 	return s.resultFromHeader(s.bc.GetHeaderByNumber(s.MessageIndexToBlockNumber(pos)))
 }
 
-func (s *ExecutionEngine) DigestMessage(num arbutil.MessageIndex, msg *arbostypes.MessageWithMetadata) error {
+func (s *ExecutionEngine) DigestMessage(num arbutil.MessageIndex, msg *arbostypes.MessageWithMetadata, msgForPrefetch *arbostypes.MessageWithMetadata) error {
```

Comment:
Why not expose another method that does prefetch instead of adding extra parameter here? Having DigestMessage do something seemingly unrelated (prefetch of next one) seems a bit counterintuitive.

If you decide to leave it as is, please add comment explaining that it does a bit more than just digesting message.

Reply:
There are couple of reason for not adding it as a separate method
1. There are many checks / input created in DigestMessage / digestMessageWithBlockMutex which are needed for prefetching 
(If we decide to move it into separate function, it might happen in future that someone modifies check for digest, but forget to modify it for prefetch.
Also there will some code repetition involved).
2. We always want to prefetch while digesting, if we have prefetch as a separate method, and if digest is called in a new place, someone might forget to call prefetch.

Also since mainly prefetch is a caching mechanism, so similar to any API, the users should not be bothered with caching outputs, so it made sense to keep prefetch tightly coupled with Digest. 

(But I do see your point around msgForPrefetch variable being confusing for the caller, so adding a comment for the function to clarify.)